doc/content/integrations/cloud-integrations/withthegrid/_index.md

```diff
@@ -0,0 +1,374 @@
+
+```tsx
+/**
+ * Source: http://www.dragino.com/downloads/downloads/LoRa_End_Node/LDS01/LDS01_LoRaWAN_Door_Sensor_UserManual_v1.2.0.pdf
+ */
+const operatingVoltageQuantityHashId: string = 'yl4wwy';
+const leakStatusQuantityHashId: string = 'dp2v9d';
+const totalLeakEventsQuantityHashId: string = 'yzrwny';
+const lastLeakDurationQuantityHashId: string = 'd4rwrd';
+
+const sensorConditionChannelIndex: number = 0;
+const leakChannelIndex: number = 1;
+
+function handle(args: Arguments, exec: Exec): Result {
+  const data = JSON.parse(args.payload);
+  if (typeof data !== 'object' || data === null) {
+    throw new Error('Data is not a JSON-object');
+  }
+  
+  const generatedAtMs = Date.parse(data.generatedAt);
+  if (Number.isNaN(generatedAtMs)) {
+    throw new Error('generatedAt cannot be parsed into a valid Date');
+  }
+  const generatedAt = new Date(generatedAtMs);
+
+  let payload = data.payload.uplink_message.decoded_payload
+  exec.addLog(JSON.stringify(typeof(payload.BAT_V)))
+
+  const operatingVoltage = payload.BAT_V;
+  const waterLeakStatus = payload.WATER_LEAK_STATUS;
+  const totalOpenLeakEvents = payload.WATER_LEAK_TIMES;
+  const lastLeakDuration = payload.LAST_WATER_LEAK_DURATION;
+
+  /**
+  * Use with raw payload
+  */
+  // const payload = data.payload;
+  // const decodedPayload = exec.base64ToUInt8Array(payload);
+  // const operatingVoltage = (decodedPayload[0] << 8 | decodedPayload[1]) & 0x3FFF;
+  // const waterLeakStatus = decodedPayload[0] & 0x40 ? 1 : 0;
+  // const totalOpenLeakEvents = decodedPayload[3] << 16 | decodedPayload[4] << 8 | decodedPayload[5];
+  // const lastLeakDuration = decodedPayload[6] << 16 | decodedPayload[7] << 8 | decodedPayload[8];
+  
+  const measurements: Result['measurements'] = [];
+
+  pushMeasurement(measurements, sensorConditionChannelIndex, operatingVoltageQuantityHashId, generatedAt, -3, operatingVoltage * 1000);
+  pushMeasurement(measurements, leakChannelIndex, leakStatusQuantityHashId, generatedAt, 0, waterLeakStatus);
+  pushMeasurement(measurements, leakChannelIndex, totalLeakEventsQuantityHashId, generatedAt, 0, totalOpenLeakEvents);
+  pushMeasurement(measurements, leakChannelIndex, lastLeakDurationQuantityHashId, generatedAt, 0, lastLeakDuration);
+ 
+  return {
+    generatedAt,
+    measurements,
+    fields: { },
+  };
+}
+
+function pushMeasurement(measurements: Result['measurements'], channelIndex: number, quantityHashId: string, generatedAt: Date, orderOfMagnitude: number, significand: number) {
+  measurements.push({
+    channelIndex: channelIndex,
+    quantityHashId: quantityHashId,
+    generatedAt: generatedAt,
+    orderOfMagnitude: orderOfMagnitude,
+    significand: significand,
+  });
+}
+```
+
+## Step 9: Powering up the device
+
+After powering up the device it should connect to the application. You can see the successful connection in the connectivity environment in **Activity** and will look like this.
```

Comment:
```suggestion
After powering up the device it should automatically connect to the Withthegrid application. The successful connection can be observed in the connectivity environment under **Activity** and it will look like this:
```